Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp

```diff
@@ -710,14 +710,8 @@ namespace AzToolsFramework
 
         void PrefabIntegrationManager::ContextMenu_SavePrefab(AZ::EntityId containerEntity)
         {
-            auto prefabPath = s_prefabPublicInterface->GetOwningInstancePrefabPath(containerEntity);
-
-            auto savePrefabOutcome = s_prefabPublicInterface->SavePrefab(prefabPath);
-
-            if (!savePrefabOutcome.IsSuccess())
-            {
-                WarningDialog("Prefab Save Error", savePrefabOutcome.GetError());
-            }
+            TemplateId templateId = s_prefabPublicInterface->GetOwningInstancePrefabTemplateId(containerEntity);
+            AZ::Interface<PrefabIntegrationManager>::Get()->m_prefabSaveHandler.ExecuteSavePrefabDialog(templateId, true);
```

Comment:
Just as mentioned above, `GetOwningInstancePrefabTemplateId` is too low level to be called directly on the PrefabIntegrationManager (which is a UI class). Consider moving this code to `m_prefabSaveHandler` so that it can be called via the entityId directly.

Reply:
just curious `AZ::Interface<PrefabIntegrationManager>::Get()` does this refer to the same class as this?